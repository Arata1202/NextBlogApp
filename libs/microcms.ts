import { createClient } from 'microcms-js-sdk';
import type {
  MicroCMSQueries,
  MicroCMSImage,
  MicroCMSDate,
  MicroCMSContentId,
} from 'microcms-js-sdk';
import { notFound } from 'next/navigation';

// カテゴリーの型定義
export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// タグの型定義
export type Tag2 = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// ライターの型定義
export type Writer = {
  name: string;
  profile: string;
  image?: MicroCMSImage;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type ContentBlock = {
  adsense?: string;
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type RelatedArticle = {
  articleLink3?: string;
};

// ブログの型定義
export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  tags?: Tag[];
  tags2?: Tag2[];
  content_blocks: ContentBlock[];
  introduction_blocks: IntroductionBlock[];
  related_articles?: RelatedArticle[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

// Initialize Client SDK.
export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

// ブログ一覧を取得
export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);
  return listData;
};

// ブログの詳細を取得
export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// カテゴリーの一覧を取得
export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

// タグの一覧を取得
export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag2>({
      endpoint: 'tags2',
      queries,
    })
    .catch(notFound);

  return listData;
};

// カテゴリーの詳細を取得
export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// タグの詳細を取得
export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag2>({
      endpoint: 'tags2',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
