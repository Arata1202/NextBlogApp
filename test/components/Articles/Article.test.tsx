import { render } from '@testing-library/react';
import ArticleComponent from '@/components/Articles/Article';
import { Article } from '@/libs/microcms';

const mockArticleData: Article = {
  id: '1',
  title: 'サンプルタイトル',
  description: 'この記事はサンプルの概要です。',
  thumbnail: { url: 'https://example.com/sample-thumbnail.jpg' },
  tags: [
    {
      id: 'tag1',
      name: 'サンプルタグ1',
      createdAt: '2023-10-01T12:00:00Z',
      updatedAt: '2023-10-01T12:00:00Z',
    },
  ],
  tags2: [
    {
      id: 'tag2',
      name: 'サンプルタグ2',
      createdAt: '2023-10-01T12:00:00Z',
      updatedAt: '2023-10-01T12:00:00Z',
    },
  ],
  introduction_blocks: [
    {
      custom_html2: '<div>サンプルのカスタムHTML</div>',
    },
  ],
  content_blocks: [
    {
      custom_html2: '<div>サンプルのカスタムHTML</div>',
    },
  ],
  related_articles: [{ articleLink3: 'https://example.com/related-article' }],
  createdAt: '2023-10-30T12:00:00Z',
  updatedAt: '2023-10-30T12:00:00Z',
  publishedAt: '2023-10-30T12:00:00Z',
};

describe('Article', () => {
  test('スナップショット（Article）', () => {
    const { asFragment } = render(<ArticleComponent data={mockArticleData} articles={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
