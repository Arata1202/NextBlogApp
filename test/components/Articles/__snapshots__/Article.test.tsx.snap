// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article スナップショット（Article） 1`] = `
<DocumentFragment>
  <div
    class="hiddenBlock categoryTitle max-w-[85rem] sm:px-6 lg:px-8 mx-auto pb-2"
  >
    <div
      class="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6"
    >
      <div
        class="lg:col-span-2"
      >
        <div
          class=""
        >
          <nav
            aria-label="Breadcrumb"
            class="flex"
          >
            <ol
              class="flex items-center space-x-4"
              role="list"
            >
              <li>
                <a
                  class="flex text-gray-500 hover:text-blue-500"
                  href="/"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 flex-shrink-0"
                    data-slot="icon"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.47 3.841a.75.75 0 0 1 1.06 0l8.69 8.69a.75.75 0 1 0 1.06-1.061l-8.689-8.69a2.25 2.25 0 0 0-3.182 0l-8.69 8.69a.75.75 0 1 0 1.061 1.06l8.69-8.689Z"
                    />
                    <path
                      d="m12 5.432 8.159 8.159c.03.03.06.058.091.086v6.198c0 1.035-.84 1.875-1.875 1.875H15a.75.75 0 0 1-.75-.75v-4.5a.75.75 0 0 0-.75-.75h-3a.75.75 0 0 0-.75.75V21a.75.75 0 0 1-.75.75H5.625a1.875 1.875 0 0 1-1.875-1.875v-6.198a2.29 2.29 0 0 0 .091-.086L12 5.432Z"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <div
                  class="flex items-center"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 flex-shrink-0 text-gray-400"
                    data-slot="icon"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16.28 11.47a.75.75 0 0 1 0 1.06l-7.5 7.5a.75.75 0 0 1-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 0 1 1.06-1.06l7.5 7.5Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <div
                    class="ml-4 text-sm font-medium text-gray-500 hover:text-blue-500"
                  />
                </div>
              </li>
              <li>
                <div
                  class="flex items-center"
                >
                  <svg
                    aria-hidden="true"
                    class="h-4 w-4 flex-shrink-0 text-gray-400"
                    data-slot="icon"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16.28 11.47a.75.75 0 0 1 0 1.06l-7.5 7.5a.75.75 0 0 1-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 0 1 1.06-1.06l7.5 7.5Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <a
                    class="ml-4 text-sm font-medium text-gray-500 hover:text-blue-500"
                    href="1"
                  >
                    サンプルタイトル
                  </a>
                </div>
              </li>
            </ol>
          </nav>
          <div
            class="space-y-5 lg:space-y-8"
          >
            <h1
              class="title text-3xl font-bold lg:text-3xl"
            >
              サンプルタイトル
            </h1>
            <picture
              class="w-full"
            >
              <source
                media="(max-width: 640px)"
                srcset="https://example.com/sample-thumbnail.jpg?fm=webp&w=414 1x, https://example.com/sample-thumbnail.jpg?fm=webp&w=414&dpr=2 2x"
                type="image/webp"
              />
              <source
                srcset="https://example.com/sample-thumbnail.jpg?fm=webp&fit=crop&w=960&h=504 1x, https://example.com/sample-thumbnail.jpg?fm=webp&fit=crop&w=960&h=504&dpr=2 2x"
                type="image/webp"
              />
              <img
                alt="サンプルタイトル"
                class="thumbnail"
                data-nimg="1"
                decoding="async"
                height="504"
                loading="eager"
                src="/_next/image?url=https%3A%2F%2Fexample.com%2Fsample-thumbnail.jpg&w=1920&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fsample-thumbnail.jpg&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fsample-thumbnail.jpg&w=1920&q=75 2x"
                style="color: transparent;"
                width="960"
              />
            </picture>
            <div
              class="FirstAd"
            />
            <div
              class="date"
            >
              <svg
                aria-hidden="true"
                class="h-5 w-5 mr-2 mt-3"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.25 12.75V12A2.25 2.25 0 0 1 4.5 9.75h15A2.25 2.25 0 0 1 21.75 12v.75m-8.69-6.44-2.12-2.12a1.5 1.5 0 0 0-1.061-.44H4.5A2.25 2.25 0 0 0 2.25 6v12a2.25 2.25 0 0 0 2.25 2.25h15A2.25 2.25 0 0 0 21.75 18V9a2.25 2.25 0 0 0-2.25-2.25h-5.379a1.5 1.5 0 0 1-1.06-.44Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
                  
            </div>
            <div
              class="date"
            />
            <div>
              <div
                class="content"
              >
                <div>
                  サンプルのカスタムHTML
                </div>
              </div>
            </div>
            <div />
            <div>
              <div>
                <div
                  class="content"
                >
                  <div>
                    サンプルのカスタムHTML
                  </div>
                </div>
              </div>
            </div>
            <div
              class="FirstAd"
            />
            <div
              class="related-articles mt-10"
            >
              <h1
                class="profile text-2xl font-semibold flex justify-center pt-10"
              >
                <svg
                  aria-hidden="true"
                  class="h-8 w-8 mr-2"
                  data-slot="icon"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.755 10.059a7.5 7.5 0 0 1 12.548-3.364l1.903 1.903h-3.183a.75.75 0 1 0 0 1.5h4.992a.75.75 0 0 0 .75-.75V4.356a.75.75 0 0 0-1.5 0v3.18l-1.9-1.9A9 9 0 0 0 3.306 9.67a.75.75 0 1 0 1.45.388Zm15.408 3.352a.75.75 0 0 0-.919.53 7.5 7.5 0 0 1-12.548 3.364l-1.902-1.903h3.183a.75.75 0 0 0 0-1.5H2.984a.75.75 0 0 0-.75.75v4.992a.75.75 0 0 0 1.5 0v-3.18l1.9 1.9a9 9 0 0 0 15.059-4.035.75.75 0 0 0-.53-.918Z"
                    fill-rule="evenodd"
                  />
                </svg>
                関連記事
              </h1>
              <div
                class="mt-5"
              >
                <div />
              </div>
            </div>
            <div
              style="padding-top: 12px;"
            >
              <h1
                class="profile text-2xl font-semibold flex justify-center mb-5"
              >
                <svg
                  aria-hidden="true"
                  class="h-8 w-8 mr-2"
                  data-slot="icon"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.493 18.5c-.425 0-.82-.236-.975-.632A7.48 7.48 0 0 1 6 15.125c0-1.75.599-3.358 1.602-4.634.151-.192.373-.309.6-.397.473-.183.89-.514 1.212-.924a9.042 9.042 0 0 1 2.861-2.4c.723-.384 1.35-.956 1.653-1.715a4.498 4.498 0 0 0 .322-1.672V2.75A.75.75 0 0 1 15 2a2.25 2.25 0 0 1 2.25 2.25c0 1.152-.26 2.243-.723 3.218-.266.558.107 1.282.725 1.282h3.126c1.026 0 1.945.694 2.054 1.715.045.422.068.85.068 1.285a11.95 11.95 0 0 1-2.649 7.521c-.388.482-.987.729-1.605.729H14.23c-.483 0-.964-.078-1.423-.23l-3.114-1.04a4.501 4.501 0 0 0-1.423-.23h-.777ZM2.331 10.727a11.969 11.969 0 0 0-.831 4.398 12 12 0 0 0 .52 3.507C2.28 19.482 3.105 20 3.994 20H4.9c.445 0 .72-.498.523-.898a8.963 8.963 0 0 1-.924-3.977c0-1.708.476-3.305 1.302-4.666.245-.403-.028-.959-.5-.959H4.25c-.832 0-1.612.453-1.918 1.227Z"
                  />
                </svg>
                シェアする
              </h1>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#00aced"
                    r="32"
                  />
                  <path
                    d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#0965FE"
                    r="32"
                  />
                  <path
                    d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#00b800"
                    r="32"
                  />
                  <path
                    d="M52.62 30.138c0 3.693-1.432 7.019-4.42 10.296h.001c-4.326 4.979-14 11.044-16.201 11.972-2.2.927-1.876-.591-1.786-1.112l.294-1.765c.069-.527.142-1.343-.066-1.865-.232-.574-1.146-.872-1.817-1.016-9.909-1.31-17.245-8.238-17.245-16.51 0-9.226 9.251-16.733 20.62-16.733 11.37 0 20.62 7.507 20.62 16.733zM27.81 25.68h-1.446a.402.402 0 0 0-.402.401v8.985c0 .221.18.4.402.4h1.446a.401.401 0 0 0 .402-.4v-8.985a.402.402 0 0 0-.402-.401zm9.956 0H36.32a.402.402 0 0 0-.402.401v5.338L31.8 25.858a.39.39 0 0 0-.031-.04l-.002-.003-.024-.025-.008-.007a.313.313 0 0 0-.032-.026.255.255 0 0 1-.021-.014l-.012-.007-.021-.012-.013-.006-.023-.01-.013-.005-.024-.008-.014-.003-.023-.005-.017-.002-.021-.003-.021-.002h-1.46a.402.402 0 0 0-.402.401v8.985c0 .221.18.4.402.4h1.446a.401.401 0 0 0 .402-.4v-5.337l4.123 5.568c.028.04.063.072.101.099l.004.003a.236.236 0 0 0 .025.015l.012.006.019.01a.154.154 0 0 1 .019.008l.012.004.028.01.005.001a.442.442 0 0 0 .104.013h1.446a.4.4 0 0 0 .401-.4v-8.985a.402.402 0 0 0-.401-.401zm-13.442 7.537h-3.93v-7.136a.401.401 0 0 0-.401-.401h-1.447a.4.4 0 0 0-.401.401v8.984a.392.392 0 0 0 .123.29c.072.068.17.111.278.111h5.778a.4.4 0 0 0 .401-.401v-1.447a.401.401 0 0 0-.401-.401zm21.429-5.287c.222 0 .401-.18.401-.402v-1.446a.401.401 0 0 0-.401-.402h-5.778a.398.398 0 0 0-.279.113l-.005.004-.006.008a.397.397 0 0 0-.111.276v8.984c0 .108.043.206.112.278l.005.006a.401.401 0 0 0 .284.117h5.778a.4.4 0 0 0 .401-.401v-1.447a.401.401 0 0 0-.401-.401h-3.93v-1.519h3.93c.222 0 .401-.18.401-.402V29.85a.401.401 0 0 0-.401-.402h-3.93V27.93h3.93z"
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#009ad9"
                    r="32"
                  />
                  <path
                    d="M 36.164062 33.554688 C 34.988281 32.234375 33.347656 31.5 31.253906 31.34375 C 33.125 30.835938 34.476562 30.09375 35.335938 29.09375 C 36.191406 28.09375 36.609375 26.78125 36.609375 25.101562 C 36.628906 23.875 36.332031 22.660156 35.75 21.578125 C 35.160156 20.558594 34.292969 19.71875 33.253906 19.160156 C 32.304688 18.640625 31.175781 18.265625 29.847656 18.042969 C 28.523438 17.824219 26.195312 17.730469 22.867188 17.730469 L 14.769531 17.730469 L 14.769531 47.269531 L 23.113281 47.269531 C 26.46875 47.269531 28.886719 47.15625 30.367188 46.929688 C 31.851562 46.695312 33.085938 46.304688 34.085938 45.773438 C 35.289062 45.148438 36.28125 44.179688 36.933594 42.992188 C 37.597656 41.796875 37.933594 40.402344 37.933594 38.816406 C 37.933594 36.621094 37.347656 34.867188 36.164062 33.554688 Z M 22.257812 24.269531 L 23.984375 24.269531 C 25.988281 24.269531 27.332031 24.496094 28.015625 24.945312 C 28.703125 25.402344 29.042969 26.183594 29.042969 27.285156 C 29.042969 28.390625 28.664062 29.105469 27.9375 29.550781 C 27.210938 29.992188 25.84375 30.199219 23.855469 30.199219 L 22.257812 30.199219 Z M 29.121094 41.210938 C 28.328125 41.691406 26.976562 41.925781 25.078125 41.925781 L 22.257812 41.925781 L 22.257812 35.488281 L 25.195312 35.488281 C 27.144531 35.488281 28.496094 35.738281 29.210938 36.230469 C 29.925781 36.726562 30.304688 37.582031 30.304688 38.832031 C 30.304688 40.078125 29.914062 40.742188 29.105469 41.222656 Z M 29.121094 41.210938 M 46.488281 39.792969 C 44.421875 39.792969 42.742188 41.46875 42.742188 43.535156 C 42.742188 45.605469 44.421875 47.28125 46.488281 47.28125 C 48.554688 47.28125 50.230469 45.605469 50.230469 43.535156 C 50.230469 41.46875 48.554688 39.792969 46.488281 39.792969 Z M 46.488281 39.792969 M 43.238281 17.730469 L 49.738281 17.730469 L 49.738281 37.429688 L 43.238281 37.429688 Z M 43.238281 17.730469 "
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#E60023"
                    r="32"
                  />
                  <path
                    d="M32,16c-8.8,0-16,7.2-16,16c0,6.6,3.9,12.2,9.6,14.7c0-1.1,0-2.5,0.3-3.7 c0.3-1.3,2.1-8.7,2.1-8.7s-0.5-1-0.5-2.5c0-2.4,1.4-4.1,3.1-4.1c1.5,0,2.2,1.1,2.2,2.4c0,1.5-0.9,3.7-1.4,5.7 c-0.4,1.7,0.9,3.1,2.5,3.1c3,0,5.1-3.9,5.1-8.5c0-3.5-2.4-6.1-6.7-6.1c-4.9,0-7.9,3.6-7.9,7.7c0,1.4,0.4,2.4,1.1,3.1 c0.3,0.3,0.3,0.5,0.2,0.9c-0.1,0.3-0.3,1-0.3,1.3c-0.1,0.4-0.4,0.6-0.8,0.4c-2.2-0.9-3.3-3.4-3.3-6.1c0-4.5,3.8-10,11.4-10 c6.1,0,10.1,4.4,10.1,9.2c0,6.3-3.5,11-8.6,11c-1.7,0-3.4-0.9-3.9-2c0,0-0.9,3.7-1.1,4.4c-0.3,1.2-1,2.5-1.6,3.4 c1.4,0.4,3,0.7,4.5,0.7c8.8,0,16-7.2,16-16C48,23.2,40.8,16,32,16z"
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#FF5700"
                    r="32"
                  />
                  <path
                    d="M 53.34375 32 C 53.277344 30.160156 52.136719 28.53125 50.429688 27.839844 C 48.722656 27.148438 46.769531 27.523438 45.441406 28.800781 C 41.800781 26.324219 37.519531 24.957031 33.121094 24.863281 L 35.199219 14.878906 L 42.046875 16.320312 C 42.214844 17.882812 43.496094 19.09375 45.066406 19.171875 C 46.636719 19.253906 48.03125 18.183594 48.359375 16.644531 C 48.6875 15.105469 47.847656 13.558594 46.382812 12.992188 C 44.914062 12.425781 43.253906 13.007812 42.464844 14.367188 L 34.625 12.800781 C 34.363281 12.742188 34.09375 12.792969 33.871094 12.9375 C 33.648438 13.082031 33.492188 13.308594 33.441406 13.566406 L 31.070312 24.671875 C 26.617188 24.738281 22.277344 26.105469 18.59375 28.609375 C 17.242188 27.339844 15.273438 26.988281 13.570312 27.707031 C 11.863281 28.429688 10.746094 30.089844 10.71875 31.941406 C 10.691406 33.789062 11.757812 35.484375 13.441406 36.257812 C 13.402344 36.726562 13.402344 37.195312 13.441406 37.664062 C 13.441406 44.832031 21.792969 50.65625 32.097656 50.65625 C 42.398438 50.65625 50.753906 44.832031 50.753906 37.664062 C 50.789062 37.195312 50.789062 36.726562 50.753906 36.257812 C 52.363281 35.453125 53.371094 33.800781 53.34375 32 Z M 21.34375 35.199219 C 21.34375 33.433594 22.777344 32 24.542969 32 C 26.3125 32 27.742188 33.433594 27.742188 35.199219 C 27.742188 36.96875 26.3125 38.398438 24.542969 38.398438 C 22.777344 38.398438 21.34375 36.96875 21.34375 35.199219 Z M 39.9375 44 C 37.664062 45.710938 34.871094 46.582031 32.03125 46.464844 C 29.191406 46.582031 26.398438 45.710938 24.128906 44 C 23.847656 43.65625 23.871094 43.15625 24.183594 42.839844 C 24.5 42.527344 25 42.503906 25.34375 42.785156 C 27.269531 44.195312 29.617188 44.90625 32 44.800781 C 34.386719 44.929688 36.746094 44.242188 38.6875 42.847656 C 39.042969 42.503906 39.605469 42.511719 39.953125 42.863281 C 40.296875 43.21875 40.289062 43.785156 39.9375 44.128906 Z M 39.359375 38.527344 C 37.59375 38.527344 36.160156 37.09375 36.160156 35.328125 C 36.160156 33.5625 37.59375 32.128906 39.359375 32.128906 C 41.128906 32.128906 42.558594 33.5625 42.558594 35.328125 C 42.59375 36.203125 42.269531 37.054688 41.65625 37.6875 C 41.046875 38.316406 40.203125 38.664062 39.328125 38.65625 Z M 39.359375 38.527344"
                    fill="white"
                  />
                </svg>
              </button>
              <button
                aria-label="シェアボタン"
                class="react-share__ShareButton m-1"
                style="background-color: transparent; padding: 0px; cursor: pointer;"
              >
                <svg
                  height="40"
                  viewBox="0 0 64 64"
                  width="40"
                >
                  <circle
                    cx="32"
                    cy="32"
                    fill="#0077B5"
                    r="32"
                  />
                  <path
                    d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
                    fill="white"
                  />
                </svg>
              </button>
            </div>
            <div
              class="mt-2"
            >
              <div
                class="pt-3"
              >
                <h1
                  class="text-2xl font-semibold flex justify-center mb-5"
                >
                  <svg
                    aria-hidden="true"
                    class="h-8 w-8 mr-2"
                    data-slot="icon"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.25 6.375a4.125 4.125 0 1 1 8.25 0 4.125 4.125 0 0 1-8.25 0ZM2.25 19.125a7.125 7.125 0 0 1 14.25 0v.003l-.001.119a.75.75 0 0 1-.363.63 13.067 13.067 0 0 1-6.761 1.873c-2.472 0-4.786-.684-6.76-1.873a.75.75 0 0 1-.364-.63l-.001-.122ZM18.75 7.5a.75.75 0 0 0-1.5 0v2.25H15a.75.75 0 0 0 0 1.5h2.25v2.25a.75.75 0 0 0 1.5 0v-2.25H21a.75.75 0 0 0 0-1.5h-2.25V7.5Z"
                    />
                  </svg>
                  フォローする
                </h1>
              </div>
              <div
                class="flex justify-center"
              >
                <a
                  aria-label="RSSフォローボタン"
                  class="bg-orange-500 rounded-full p-2 m-1"
                  href="https://meta.realunivlog.com/rss.xml"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6 text-white"
                    data-slot="icon"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3.75 4.5a.75.75 0 0 1 .75-.75h.75c8.284 0 15 6.716 15 15v.75a.75.75 0 0 1-.75.75h-.75a.75.75 0 0 1-.75-.75v-.75C18 11.708 12.292 6 5.25 6H4.5a.75.75 0 0 1-.75-.75V4.5Zm0 6.75a.75.75 0 0 1 .75-.75h.75a8.25 8.25 0 0 1 8.25 8.25v.75a.75.75 0 0 1-.75.75H12a.75.75 0 0 1-.75-.75v-.75a6 6 0 0 0-6-6H4.5a.75.75 0 0 1-.75-.75v-.75Zm0 7.5a1.5 1.5 0 1 1 3 0 1.5 1.5 0 0 1-3 0Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </a>
                <a
                  aria-label="Feedlyフォローボタン"
                  class="bg-green-500 rounded-full p-2 m-1"
                  href="https://feedly.com/i/subscription/feed/https://meta.realunivlog.com/rss.xml"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6 text-white"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.85995 1.98852a2.60906 2.60906 0 00-3.72608 0L.76766 11.52674a2.72906 2.72906 0 000 3.79508l6.68415 6.80816a2.61806 2.61806 0 001.74004.66401h5.61313a2.61606 2.61606 0 001.87204-.79101l6.55415-6.67516a2.72606 2.72606 0 000-3.79508l-9.37021-9.54422zm-.26 17.4224l-.93502.95002a.372.372 0 01-.268.114h-.80003a.376.376 0 01-.247-.096l-.95402-.97002a.39.39 0 010-.54201l1.33703-1.36003a.37.37 0 01.531 0l1.33704 1.36103a.389.389 0 010 .543zm0-5.71113l-3.73709 3.80808a.374.374 0 01-.268.111h-.79902a.376.376 0 01-.25-.093l-.95103-.97002a.391.391 0 010-.54401l4.1391-4.2141a.372.372 0 01.531 0l1.33704 1.36204a.386.386 0 010 .54zm0-5.70713L7.0598 14.6528a.372.372 0 01-.268.113h-.80002a.373.373 0 01-.24901-.094l-.95302-.97202a.388.388 0 010-.54001L11.7329 6.0896a.372.372 0 01.531 0l1.33704 1.36004a.389.389 0 010 .543z"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="flex justify-center mt-5"
              >
                <a
                  href="https://blogmura.com/profiles/11190305/?p_cid=11190305&reader=11190305"
                  target="_blank"
                >
                  <img
                    alt="リアル大学生 - にほんブログ村"
                    height="36"
                    loading="lazy"
                    src="https://b.blogmura.com/banner-blogmura-reader-white-small.svg"
                    width="160"
                  />
                </a>
                <a
                  class="ml-3"
                  href="https://blog.with2.net/link/?id=2117761"
                  target="_blank"
                  title="人気ブログランキング"
                >
                  <img
                    alt="人気ブログランキング"
                    height="36"
                    loading="lazy"
                    src="https://blog.with2.net/img/banner/banner_22.gif"
                    width="93"
                  />
                </a>
                <a
                  class="ml-3"
                  href="https://blogranking.fc2.com/in.php?id=1067087"
                  target="_blank"
                >
                  <img
                    loading="lazy"
                    src="https://static.fc2.com/blogranking/ranking_banner/a_02.gif"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="articleSidebar"
      />
    </div>
  </div>
</DocumentFragment>
`;
